core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -543,7 +543,7 @@ private[spark] class BlockManager(
    */
   override def getBlockData(blockId: BlockId): ManagedBuffer = {
     if (blockId.isShuffle) {
-      shuffleManager.shuffleBlockResolver.getBlockData(blockId.asInstanceOf[ShuffleBlockId])
+      shuffleManager.shuffleBlockResolver.getBlockData(-1, blockId.asInstanceOf[ShuffleBlockId])
```

Comment:
we should add a comment saying that, the old shuffle protocol will hit this branch.

Reply:
Thanks for the advice, done in 2362512.